### DISCUSSION TITLE
New JSON syntax for HAA after update - JSON for 4CH Pro r2 not working anymore

### DISCUSSION THREAD

**Discussion by asbesto99** [RavenSystem/esp-homekit-devices#2129] [MEPLHAA Scripts]
Good evening all,

I used to have my old JSON to manage a Sonoff 4CH Pro R2, to have 1 lock, 1 switch, 2 garage doors.
Now I understand that the syntax with new HAA version is different so I cannot use my old JSON anymore:

{"c":
{"l":13,"b":[{"g":0,"t":5}]},
"a":[
{
"t":4,
"i":5,
"0":{"r":[{"g":15,"v":1,"i":0.4}]},
"1":{"r":[{"g":15}]},
"b":[{"g":14}]
},
{
"t":40,
    "d":35,
    "c":40,
    "0":{"r":[{"g":12,"v":1,"i":0.5}]},
    "1":{"a":0},
    "2":{"r":[{"g":12,"i":1.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},
    "3":{"a":2}
},
{
"t":40,
    "d":15,
    "c":23,
    "0":{"r":[{"g":5,"v":1,"i":0.5}]},
    "1":{"a":0},
    "2":{"r":[{"g":5,"i":1.5},{"g":5,"v":1,"i":0.5},{"g":5,"v":1,"i":2}]},
    "3":{"a":2}
},
{"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1}]},"b":[{"g":10}]}
]}

So I am looking for some help to achieve having:

Channel 1: switch
Channel 2: garage door
Channel 3: garage door 2
Channel  4: Lock

Any help you may share will be more than appreciated, thanks!

**Comment by asbesto99** [comment]
@RavenSystem 

hola Josè I believe you are the only one able to rewrite this (it was originally written by you) :)

any chances you could send some help?

thanks!

**Comment by RavenSystem** [comment]
```json
{"c":
{"io":[[[4,5,12,15,13],2],[[0],6],[[10,14],6,1]],"l":13,"b":[[0,5]]},
"a":[
{
"t":4,
"i":5,
"0":{"r":[[15,1,0.4]]},
"1":{"r":[[15]]},
"b":[[14]]
},
{
"t":40,
"d":35,
"c":40,
"0":{"r":[[12,1,0.5]]},
"1":{"a":0},
"2":{"r":[[12,0,1.5],[12,1,0.5],[12,1,2]]},
"3":{"a":2}
},
{
"t":40,
"d":15,
"c":23,
"0":{"r":[[5,1,0.5]]},
"1":{"a":0},
"2":{"r":[[5,0,1.5],[5,1,0.5],[5,1,2]]},
"3":{"a":2}
},
{"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[10]]}
]}
```

**Comment by asbesto99** [comment]
Hi Jose,

your JSON is:

Channel 1: garage door 2
Channel 2: garage door
Channel 3: switch
Channel 4: Lock

What I need is: 

Channel 1: switch
Channel 2: garage door
Channel 3: garage door 2
Channel 4: Lock

could
You please modify it? Thanks!
